src/utils/exporters.ts

```diff
@@ -1,59 +1,95 @@
 import { Annotation, ANNOTATION_COLUMNS, GENE_COLUMNS, StructuredData } from './ingest';
+import { Readable, PassThrough } from "stream"
+import combine from "combine-streams";
 
-const column_modifiers: Map<string|null, json2csv.FieldInfo<Annotation>> = new Map([
+const column_modifiers: Map<string | null, json2csv.FieldInfo<Annotation>> = new Map([
     [null, {
         value: () => ''
     }],
     ["Invert", {
         value: (row: Annotation) => row.Invert === true ? "NOT" : ''
     }],
     ["AlternativeGeneName", {
-        value: (row: Annotation) => (row.AlternativeGeneName||[]).join("|")
+        value: (row: Annotation) => (row.AlternativeGeneName || []).join("|")
     }],
     ["AdditionalEvidence", {
-        value: (row: Annotation) => (row.AdditionalEvidence||[]).join("|")
+        value: (row: Annotation) => (row.AdditionalEvidence || []).join("|")
     }],
     ["Date", {
         // return new Date(Date.parse(`${value.slice(0,4)}-${value.slice(4,6)}-${value.slice(6,8)}`));
-        value: (row) => new Date(row.Date).toJSON().split("T")[0].replace(/\-/g,"")
+        value: (row) => new Date(row.Date).toJSON().split("T")[0].replace(/\-/g, "")
     }]
 ])
 
-const metadataSerializer = (metadata: {[key: string]: string}) => Object.entries(metadata)
-  .map((([key, value]) => `!${key}: ${value}`))
-  .reduce((accum, curr) => `${accum}\n${curr}`, "");
+const metadataSerializer = (metadata: { [key: string]: string }) => Object.entries(metadata)
+    .map((([key, value]) => `!${key}: ${value}`))
+    .reduce((accum, curr) => `${accum}\n${curr}`, "");
 
-export function annotationsToGAF(data: StructuredData, additionalMetadata: {[key: string]: string} = {}){
-    const serialized = serialize(data.annotations.records, {
-        fields: ANNOTATION_COLUMNS.map(col=>column_modifiers.get(col) || col as string),
+export function annotationsToGAF(data: StructuredData, additionalMetadata: { [key: string]: string } = {}) {
+    const transformer = new Transform({
+        fields: ANNOTATION_COLUMNS.map(col => column_modifiers.get(col) || col as string),
         header: false,
         defaultValue: "",
         excelStrings: false,
         quote: '',
         delimiter: '\t',
-    });
+    }, { objectMode: true })
+    // const serialized = serialize(data.annotations.records, );
+    //@ts-ignore
+    transformer._implicitHeader = () => {};
 
-    const header = Object.keys(additionalMetadata).length>0 ? 
-      `${data.annotations.metadata.trim()}\n${metadataSerializer(additionalMetadata).trim()}` : 
-      data.annotations.metadata
+    const header = Object.keys(additionalMetadata).length > 0 ?
+        `${data.annotations.metadata.trim()}\n${metadataSerializer(additionalMetadata).trim()}` :
+        data.annotations.metadata
 
-    return header.concat("\n",serialized);
+    const input = new Readable({ objectMode: true });
+    input._read = () => { };
+
+    for (const record of data.annotations.records) {
```

Comment:
Correct me if I'm wrong but this means that you're serializing _all_ of the data that's been parsed, not just a queried subset, right?